app/models/server/models/Subscriptions.js

```diff
@@ -1429,6 +1429,25 @@ export class Subscriptions extends Base {
 
 		return this.update(query, update);
 	}
+
+	deleteAllThreadsByRoomId(rid, ts, users) {
+		const query = {
+			rid,
+			ts,
```

Comment:
you cannot query the same `ts` as queried on `Messages` because they don't "match".. a message's `ts` is the timestamp when that message was sent and a subscription's `ts` is when the subscription was **created** (i.e.: the user joined the room)